pkg/segment/reader/segread/segstatsreader.go

```diff
@@ -188,113 +188,139 @@ func readSingleSst(fdata []byte, qid uint64) (*structs.SegStats, error) {
 }
 
 func GetSegMin(runningSegStat *structs.SegStats,
-	currSegStat *structs.SegStats) (*utils.NumTypeEnclosure, error) {
+	currSegStat *structs.SegStats) (*utils.CValueEnclosure, error) {
 
-	rSst := utils.NumTypeEnclosure{
-		Ntype:    utils.SS_DT_SIGNED_NUM,
-		IntgrVal: math.MaxInt64,
-	}
+	result := utils.CValueEnclosure{}
 
 	if currSegStat == nil {
 		log.Errorf("GetSegMin: currSegStat is nil")
-		return &rSst, errors.New("GetSegMin: currSegStat is nil")
-	}
-
-	if !currSegStat.IsNumeric {
-		log.Errorf("GetSegMin: current segStats is non-numeric")
-		return &rSst, errors.New("GetSegMin: current segStat is non-numeric")
+		return &result, errors.New("GetSegMin: currSegStat is nil")
 	}
 
 	// if this is the first segment, then running will be nil, and we return the first seg's stats
 	if runningSegStat == nil {
+		if !currSegStat.IsNumeric && currSegStat.StringStats.Min.Dtype == utils.SS_DT_STRING {
+			return &utils.CValueEnclosure{
+				Dtype: utils.SS_DT_STRING,
+				CVal:  currSegStat.StringStats.Min.CVal,
+			}, nil
+		}
+
 		switch currSegStat.NumStats.Min.Ntype {
 		case utils.SS_DT_FLOAT:
-			rSst.FloatVal = currSegStat.NumStats.Min.FloatVal
-			rSst.Ntype = utils.SS_DT_FLOAT
+			result.Dtype = utils.SS_DT_FLOAT
+			result.CVal = currSegStat.NumStats.Min.FloatVal
 		default:
-			rSst.IntgrVal = currSegStat.NumStats.Min.IntgrVal
+			result.Dtype = utils.SS_DT_SIGNED_NUM
+			result.CVal = currSegStat.NumStats.Min.IntgrVal
 		}
-		return &rSst, nil
+
+		return &result, nil
+	}
+
+	if !currSegStat.IsNumeric {
+		runningSegStat.StringStats.MergeMinStrStats(currSegStat.StringStats)
+
+		if runningSegStat.StringStats.Min.Dtype == utils.SS_DT_STRING {
```

Comment:
Will we ever get to line 224 and have the condition be false?

Reply:
It might if there is no data in the currSegStat and runningSegStat or there was an error which caused these fields to be not initialized and we would just want to return INVALID CValEnclosure.